pkg/controller/deployment/deployment_controller.go

```diff
@@ -170,6 +174,25 @@ func (r *ReconcileDeployment) Reconcile(request reconcile.Request) (reconcile.Re
 	return reconcile.Result{}, nil
 }
 
+func removeSideCar(r *ReconcileDeployment, dep *appsv1.Deployment, ctx context.Context) {
```

Comment:
Isn't it better to make `removeSidecar` to be an instance method, instead of receiving an instance as a parameter? Also, Go conventions dictate that the context should be the first parameter. This would be the final signature:

```suggestion
func (r *ReconcileDeployment) removeSidecar(ctx context.Context, dep *appsv1.Deployment) {
```